src/core/Particles.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react'
+import * as THREE from 'three'
+import { PointsProps, useFrame, extend, Node } from '@react-three/fiber'
+import { shaderMaterial } from '.'
+
+// eslint-disable-next-line
+// @ts-ignore
+import fragShader from '../helpers/glsl/Particles.frag.glsl'
+// eslint-disable-next-line
+// @ts-ignore
+import vertShader from '../helpers/glsl/Particles.vert.glsl'
+
+interface Props {
+  count?: number
+  speed?: number
+  opacity?: number
+  color?: THREE.ColorRepresentation
```

Comment:
Also consider having color as an attribute eg.  https://drei-git-fork-farazzshaikh-feat-particles-pmndrs.vercel.app/?path=/story/performance-points--basic-points-buffer

Reply:
Acutally, just attributes all the things

```jsx
speed/opacity/color?: number | bufferAttribute
```

if the user passes a single value => use it to fill the attribute, otherwise use the passed attribute.

This way you can do all sorts of fun things with this and it doesn't really explode complexity